### PR TITLE
Add shieldhit executable mock resolves #639

### LINKED ISSUE #639: add Mock executable for SHIELD-HIT12A
Similar as https://github.com/DataMedSci/pymchelper/issues/637

### PR DESCRIPTION
**Pull request by hendzeld**

### CODE REVIEW

**Comment by grzanka** [comment]
@hendzeld fix conflict

File: tests/res/mocks/libs/bash_script.py [hunk 1]
```diff
@@ -43,12 +45,12 @@ def generate_mock(output_path: Path, files_to_save: List[Path], stdout: Optional
...
-                script.write(__STDOUT_TEMPLATE.format(stdout=stdout_content))
+                base64_encoded_content = base64.standard_b64encode(stdout_content).decode("utf-8")
+                script.write(__STDOUT_TEMPLATE.format(stdout=base64_encoded_content))
         if stderr:
             with open(stderr, 'rb') as f:
                 stderr_content = f.read()
-                stderr_content = base64.standard_b64encode(stderr_content).decode("utf-8")
-                script.write(__STDERR_TEMPLATE.format(stderr=stderr_content))
+                base64_encoded_content = base64.standard_b64encode(stderr_content).decode("utf-8")
+                script.write(__STDERR_TEMPLATE.format(stderr=base64_encoded_content))
```
Comment (grzanka) [inline tests/res/mocks/libs/bash_script.py:54]: /reviewpad explain

Comment (hendzeld) [reply tests/res/mocks/libs/bash_script.py:54]: I added another variable. The previous one was of type bytes and this one is a string (utf-8 and base64 encoded representation of stdout file)

```
            with open(stdout, 'rb') as f:
                # stdout as bytes
                stdout_content = f.read()
                # stdout as utf-8 encoded base64 string
                base64_encoded_content = base64.standard_b64encode(stderr_content).decode("utf-8")
                script.write(__STDOUT_TEMPLATE.format(stdout=base64_encoded_content))
```

I can undo this change if it doesn't add value, or perhaps rename **stdout_content** to **stdout_byte**

File: tests/res/mocks/shieldhit_minimal/detect.dat [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Geometry Cyl
+    Name ScoringCylinder
+        R 0.0 10.0 4
+        Z 0.0 30.0 4
```
Comment (grzanka) [inline tests/res/mocks/shieldhit_minimal/detect.dat:4]: Better to use cartesian mesh divived into 4x4 bins, dividing cylinder along R and along Z is quite rare.

File: tests/res/mocks/libs/bash_script.py (hunk 1 above)
Comment (grzanka) [reply tests/res/mocks/libs/bash_script.py:54]: @hendzeld that is OK, I was just testing reviewpad capabilities

**Review by grzanka — APPROVED** [review]

**Pull request merged**